mindsdb/integrations/handlers/gcs_handler/gcs_handler.py

```diff
@@ -0,0 +1,360 @@
+from contextlib import contextmanager
+
+import json
+import duckdb
+import pandas as pd
+import fsspec
+import google.auth
+from google.cloud import storage
+from typing import Text, Dict, Optional, List
+from duckdb import DuckDBPyConnection
+
+from mindsdb_sql.parser.ast.base import ASTNode
+from mindsdb_sql.parser.ast import Select, Identifier, Insert
+
+from mindsdb.utilities import log
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+from mindsdb.integrations.libs.api_handler import APIResource, APIHandler
+from mindsdb.integrations.utilities.sql_utils import FilterCondition
+
+logger = log.getLogger(__name__)
+
+
+class ListFilesTable(APIResource):
+
+    def list(self,
+             targets: List[str] = None,
+             conditions: List[FilterCondition] = None,
+             *args, **kwargs) -> pd.DataFrame:
+
+        tables = self.handler._get_tables()
+        data = []
+        for path in tables:
+            path = path.replace('`', '')
+            item = {
+                'path': path,
+                'name': path[path.rfind('/') + 1:],
+                'extension': path[path.rfind('.') + 1:]
+            }
+
+            data.append(item)
+
+        return pd.DataFrame(data=data, columns=self.get_columns())
+
+    def get_columns(self) -> List[str]:
+        return ["path", "name", "extension", "content"]
+
+
+class FileTable(APIResource):
+
+    def list(self, targets: List[str] = None, table_name=None, *args, **kwargs) -> pd.DataFrame:
+        return self.handler._read_as_table(table_name)
+
+    def add(self, data, table_name=None):
+        df = pd.DataFrame(data)
+        return self.handler._add_data_to_table(table_name, df)
+
+
+class GcsHandler(APIHandler):
+    """
+    This handler handles connection and execution of the SQL statements on GCS.
+    """
+
+    name = 'gcs'
+
+    supported_file_formats = ['csv', 'tsv', 'json', 'parquet']
+
+    def __init__(self, name: Text, connection_data: Optional[Dict], **kwargs):
+        """
+        Initializes the handler.
+
+        Args:
+            name (Text): The name of the handler instance.
+            connection_data (Dict): The connection data required to connect to the GCS account.
+            kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+        self.is_select_query = False
+        self.table_name = None
```

Comment:
These instance do not seem to be in use anymore - can we please remove them?

Reply:
Removed it.